lib/core/Manager.js

```diff
@@ -124,7 +125,7 @@ Manager.prototype.resolve = function () {
 
 Manager.prototype.preinstall = function (json) {
     var that = this;
-    var componentsDir = path.join(this._config.cwd, this._config.directory);
```

Comment:
When the config.directory begins with an `/` it's expected to be absolute. In any other case the path is expected to be relative and joined with `config.cwd`. Not sure if I understood your comment correctly. How can `path.resolve` help here?